src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -26,31 +26,52 @@ public DiagnosticsHandler(HttpMessageHandler innerHandler) : base(innerHandler)
 
         internal static bool IsEnabled()
         {
-            //check if someone listens to HttpHandlerDiagnosticListener
-            return s_diagnosticListener.IsEnabled();
+            // check if there is a parent Activity (and propagation is not suppressed)
+            // or if someone listens to HttpHandlerDiagnosticListener
+            return EnableActivityPropagation && (Activity.Current != null || s_diagnosticListener.IsEnabled());
```

Comment:
We've paid relatively little attention to the perf of DiagnosticHandler because it was opt-in based on enabling tracing and the like. This is now going to make it on much of the time, right? What kind of analysis/work is being done to measure/mitigate the overhead?

Reply:
It'll realistically only affect ASP.NET Core scenarios with this change since the intent was to flow if an existing Activity exists (which is now the default). Compared to making an outbound HTTP request, the Activity should be negligible but that needs to be measured end to end (and you can always set the activity to null).   

As an aside, ASP.NET Core isn't good with logging turned on (which now turns on Activities as well) https://github.com/aspnet/AspNetCore/issues/9594.

Reply:
Ok. Thanks.  Glad to hear it's being tracked.